engine/src/main/java/org/hibernate/validator/internal/cfg/context/DefaultConstraintMapping.java

```diff
@@ -83,4 +91,31 @@ public final AnnotationProcessingOptionsImpl getAnnotationProcessingOptions() {
 
 		return configurations;
 	}
+
+	@Override
+	public <A extends Annotation> ConstraintDefinitionContext<A> constraint(Class<A> annotationClass) {
+		Contracts.assertNotNull( annotationClass, MESSAGES.annotationTypeMustNotBeNull() );
+		Contracts.assertTrue( annotationClass.isAnnotationPresent( Constraint.class ),
```

Comment:
Interesting, i'd expect such check (in a more complete fashion) to happen later in the pipeline anyways? It doesn't do harm here, but it seems not strictly needed?